src/xz/mod.rs

```diff
@@ -0,0 +1,120 @@
+//! Logic for handling `.xz` file format.
+//!
+//! Format specifications are at [https://tukaani.org/xz/xz-file-format.txt](spec).
+//!
+//! [spec]: https://tukaani.org/xz/xz-file-format.txt
+
+#![deny(missing_docs)]
+#![deny(missing_debug_implementations)]
+
+use crate::error;
+use std::io;
+
+pub(crate) mod footer;
+pub(crate) mod header;
+
+/// Stream flags, see sect. 2.1.1.2.
+///
+/// This does not store the leading null byte, which is currently unused.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub(crate) struct StreamFlags {
+    pub(crate) check_method: CheckMethod,
+}
+
+impl StreamFlags {
+    /// Parse Stream Flags from a 16bits value.
+    pub(crate) fn parse(input: u16) -> error::Result<Self> {
+        let flags_bytes = input.to_be_bytes();
+
+        if flags_bytes[0] != 0x00 {
+            return Err(error::Error::XZError(format!(
+                "Invalid null byte in Stream Flags: {:x}",
+                flags_bytes[0]
+            )));
+        }
+
+        let flags = Self {
+            check_method: CheckMethod::try_from(flags_bytes[1])?,
+        };
+        Ok(flags)
+    }
+
+    /// Serialize Stream Flags into a writer.
+    pub(crate) fn serialize<W>(self, writer: &mut W) -> io::Result<usize>
```

Comment:
@gendx I was unsure whether you wanted a `to_u16` here, or this kind of directly writing method.

Reply:
It's fine within the `StreamFlags` abstraction.